lib/provisioning/ProvisionDispatcher.js

```diff
@@ -76,6 +76,22 @@ class ProvisionDispatcher {
         ], cb);
     }
 
+    checkState(cb, autoTrigger) {
+        this._onLeaderChange(autoTrigger);
+        this._onOwnerChange(autoTrigger);
+        this._onMyselfChange(cb, autoTrigger);
+        this._onProvisionChange(autoTrigger);
+        return undefined;
+    }
+
+    periodicCheckState(cb) {
+        setTimeout(() => {
+            this._log.debug('periodic check state');
+            this.checkState(cb, false);
```

Comment:
I am under the impression that this call will accumulate watchers for each `getChildren` call that happen, but never release the watcher if there's no change. If that is correct, it is an important issue that needs a fix, maybe by having the watcher of the zookeeper calls only passed when `autoTrigger` is true, instead of re-calling the `onFoo` functions only when `autoTrigger` is true?

Reply:
yes you are right

Reply:
Fixed it